scRNA-seq-advanced/05-aucell.Rmd

```diff
@@ -0,0 +1,403 @@
+---
+title: "Pathway Analysis: AUCell"
+output:
+  html_notebook:
+    toc: true
+    toc_float: true
+author: CCDL for ALSF
+date: 2024
+---
+
+*Adapted from [the AUCell vignette](https://bioconductor.org/packages/release/bioc/vignettes/AUCell/inst/doc/AUCell.html) and [the `cell-type-ewings` module](https://github.com/AlexsLemonade/OpenScPCA-analysis/tree/main/analyses/cell-type-ewings) that is part of the Open Single-cell Pediatric Cancer Atlas project.*
+
+## Objectives
+
+- Introduce the `AUCell` R package
+- Illustrate how AUC values are calculated
+- Demonstrate how AUC values can be used for cell assignment and plotting
+
+---
+
+In this notebook, we'll demonstrate how to use the AUCell method, introduced in [Aibar _et al_. 2017.](https://doi.org/10.1038/nmeth.4463), to calculate numeric values that estimate the activity of a gene set in an individual cell.
+
+AUCell calculates the area under the recovery curve (AUC), which "represents the proportion of expressed genes in the signature and their relative expression value compared to the other genes within the cell" ([Aibar _et al_. 2017.](https://doi.org/10.1038/nmeth.4463)).
+
+We will use an snRNA-seq of a Ewing sarcoma sample from the [`SCPCP000015` project](https://scpca.alexslemonade.org/projects/SCPCP000015) on the Single-cell Pediatric Cancer Atlas Portal and two relevant gene sets from the Molecular Signatures Database (MSigDB) to demonstrate this method.
+
+## Set up
+
+### Libraries
+
+```{r libraries}
+# We will be loading a SingleCellExperiment object into our environment but 
+# don't need to see the startup messages
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+})
```

Comment:
Minor, but I noticed that the two lines with comments are broken up by a message in the rendered HTML which seems weird. I might make that comment on one line to avoid that?